### DISCUSSION TITLE
Wizard submit on each step and update state

### DISCUSSION THREAD

**Discussion by CSTDev** [data-driven-forms/react-forms#1341] [Need help]
Hi,

I'm trying to use the wizard and on each step submit the data so far to a backend, take the response and put that into the form state as some fields will have been updated. I'm not entirely sure where to start, any pointers would be much appreciated?

**Comment by rvsia** [comment]
Hello,

I would start with `getRegisteredFields` and `getState().values`: https://data-driven-forms.org/hooks/use-form-api#getregisteredfields

Using `getRegisteredFields` you can get currently shown fields and with using some `get` function you can extract their values from `getState().values`.

**Comment by Hyperkid123** [comment]
Hello @CSTDev To me it almost sounds like if you want each of the wizard step to be its "own form". If that is the case, it might be actually the best approach. Handle each step separately and "fake" the wizard layout.
